### PR TITLE
Custom inputs for updating RPC URLs

### PR DESCRIPTION
**Pull request by hkirat**
Closes https://github.com/coral-xyz/backpack/issues/941 

Awaiting review from design. 

### Solana
Screenshot 2022-10-06 at 2 18 12 AM
Screenshot 2022-10-06 at 2 37 08 AM
Screenshot 2022-10-06 at 2 37 15 AM
Screenshot 2022-10-06 at 2 37 25 AM

### Eth
Screenshot 2022-10-06 at 2 37 37 AM

Screenshot 2022-10-06 at 2 37 54 AM
Screenshot 2022-10-06 at 2 37 43 AM

### CODE REVIEW

**Comment by tomlinton** [comment]
This might be a separate PR, but it'd be nice if we the Custom option allowed you to persist these into a list that showed up below (effectively allowing multiple custom RPC settings with the ability to quickly switch between them). This is something that is super useful as a developer, but might not be useful to anyone else.

**Comment by hkirat** [comment]
@tomlinton I think that makes a lot of sense. 
Just so I understand this correctly, you mean we - 
1. Store these custom RPCs locally
2. If a user has more than 0 of these stored locally, we show these as a list below the currently available 4 options as a `Recently used` set of RPCs?

**Comment by tomlinton** [comment]
Yea, and `Custom` is an `Add Custom` button, and you can delete/edit things in the custom list. We can punt this for now though, as what you've got should cover most use cases.

File: packages/app-extension/src/components/Unlocked/Settings/Preferences/Ethereum/CustomRpcUrl.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            </ListItem>
+            <ListItem
+              button={false}
+              isLast
+              style={{
+                height: "44px",
+                padding: "10px",
+              }}
+            >
+              <Typography style={{ width: "80px" }}>Chian ID</Typography>
```
Comment (0xInq) [inline packages/app-extension/src/components/Unlocked/Settings/Preferences/Ethereum/CustomRpcUrl.tsx:107]: just a typo, this should be `Chain ID`.

**Comment by hkirat** [comment]
cc @trucy @aarongillett for design review

**Comment by trucy** [comment]
Link to Figma with input states in light and dark mode: https://www.figma.com/file/HNF2HJZYq6dUzCWxSmH2Oe/Backpack-_-Questlog?node-id=12036%3A44072

Related issue that would be fixed with this work is https://github.com/coral-xyz/backpack/issues/880

**Comment by hkirat** [comment]
Updated as per designs to close #880 along with this one. 
@trucy you can find the new extension attached for review
[dev.zip](https://github.com/coral-xyz/backpack/files/9795790/dev.zip)

Have made changes across a lot of files to use the new `input styles`

**Comment by tomlinton** [comment]
Is there a need to add a separate TextInput component here, was it for styling reasons? I think in general we want to move all our components to those from `react-xnft-renderer` so that the design is consistent across the wallet and xNFTs.

**Comment by hkirat** [comment]
So after https://github.com/coral-xyz/backpack/pull/1023 gets merged in the `react-xnft-renderer` gets purged and all components move to `react-xnft-dom-renderer`.
The `react-xnft-dom-renderer` would probably be moved to use `tailwind`, and maybe even support the latest version of react so we might not be able to share components across the core app and the xnft renders. 
I think this is by design since we want to move fast on the `xnft renderer` and provide a lot of flexibility there. 

So as you'll see in https://github.com/coral-xyz/backpack/pull/1023 I separated out the `Components.tsx` from `react-xnft-renderer` to `app-extensions` because of the same reason (They're basically duplicated right now, but the former would probably diverge out to use tailwind)

**Comment by tomlinton** [comment]
Ahh great, makes sense!

**Comment by hkirat** [comment]
cc @trucy for review

**Comment by trucy** [comment]
Hey @hkirat thanks for the update! I downloaded the attachment and ran it but I'm not sure if I'm seeing the changes across inputs. I still see the focus/active state border showing up as Base/900 (#030A19), when it should be (#787C89) in a few places
Screen Shot 2022-10-25 at 3 27 18 PM
Screen Shot 2022-10-25 at 3 26 03 PM

I also noticed the top two corners of the grouped input for Change RPC Connection look square for both Ethereum and Solana
Screen Shot 2022-10-25 at 3 29 46 PM
Screen Shot 2022-10-25 at 3 29 28 PM

**Comment by hkirat** [comment]
Great 👁️ @trucy, added input styles in the places you mentioned @trucy . Unfortunately there was no centralised place for all these inputs until now so these got missed (and there might be other places where it still got missed, although I've tried to manually go through all paths to ensure there aren't). The good thing is that now there's a centralised place so any future style changes should be propagated to all the places.

Screenshot 2022-10-30 at 3 14 16 AM


Screenshot 2022-10-30 at 3 29 02 AM

Fixed the borders you mentioned above as well 
Screenshot 2022-10-30 at 3 28 37 AM
Screenshot 2022-10-30 at 3 28 28 AM

Might merge this in soon since it's becoming a little big and goes out of date with master very quickly, but will follow up with any places the styles might've gotten missed.

**Pull request merged**